### PR TITLE
Phan fixes

### PR DESCRIPTION
**Pull request by rullzer**
Me and @LukasReschke have been working on getting a stricter phan config. This is the first part of fixes to get that done.

Review should be straight forward. But probably easiest to go trough them commit by commit.

### CODE REVIEW

File: lib/private/Server.php [hunk 1]
```diff
@@ -1247,7 +1247,6 @@ public function getConfig() {
 	}
 
 	/**
-	 * @internal For internal use only
```
Comment (MorrisJobke) [inline lib/private/Server.php]: Why? This is only a private namespace interface. For external use IConfig should be used.

File: settings/Controller/UsersController.php [hunk 2]
```diff
@@ -113,7 +112,7 @@ class UsersController extends Controller {
 	 * @param string $appName
 	 * @param IRequest $request
 	 * @param IUserManager $userManager
-	 * @param IGroupManager $groupManager
+	 * @param \OC\Group\Manager $groupManager
```
Comment (MorrisJobke) [inline settings/Controller/UsersController.php]: What is the background here? We should try to also use public interfaces in our internal apps.

Comment (rullzer) [inline settings/Controller/UsersController.php]: Probabaly for the subadmin thingy

File: lib/private/Server.php (hunk 1 above)
Comment (rullzer) [inline lib/private/Server.php]: Maybe phan is to strict here...
But this is already private. So no need to add the internal flag to every function.

File: lib/composer/composer/autoload_classmap.php [hunk 3]
```diff
@@ -6,6 +6,69 @@
 $baseDir = dirname(dirname($vendorDir));
 
 return array(
+    'DeepCopy\\DeepCopy' => $vendorDir . '/myclabs/deep-copy/src/DeepCopy/DeepCopy.php',
```
Comment (LukasReschke) [inline lib/composer/composer/autoload_classmap.php]: I'd prefer if we not commit the phan stuff.

**Review by LukasReschke — DISMISSED** [review]

File: lib/composer/composer/autoload_classmap.php (hunk 3 above)
Comment (rullzer) [inline lib/composer/composer/autoload_classmap.php]: Aaah I thought I removed that... Let me take care

**Review by MorrisJobke — APPROVED** [review]

**Review by icewind1991 — APPROVED** [review]

**Comment by MorrisJobke** [comment]
```
There were 57 errors:
```


🙈 

And please add the sign-offs ;)

**Comment by MorrisJobke** [comment]
```
There were 4 failures:
```

and a rebase is needed.

🙈 🙉 🙊

**Pull request merged**